src/test/java/catan/services/impl/PlayServiceImplTest.java

```diff
@@ -756,22 +885,220 @@ public void shouldUpdateVictoryPointsOnMultipleBuildingsInMainStage() throws Exc
         //GIVEN
         hex_0_0.getEdges().getTopRight().setBuilding(new Building<EdgeBuiltType>(EdgeBuiltType.ROAD, gameUser1));
         game.setStage(GameStage.MAIN);
+        game.setDiceThrown(true);
         when(gameDao.getGameByGameId(1)).thenReturn(game);
 
         // WHEN
         assertEquals(0, gameUser1.getAchievements().getDisplayVictoryPoints());
 
         allowUserToBuildSettlement(gameUser1);
-        playService.buildSettlement(gameUser1.getUser(), "1", "3");
+        Map<String, String> params = new HashMap<String, String>();
+        params.put("nodeId", "3");
+
+        playService.processAction(GameUserActionCode.BUILD_SETTLEMENT, gameUser1.getUser(), "1", params);
 
         assertEquals(1, gameUser1.getAchievements().getDisplayVictoryPoints());
 
         allowUserToBuildCity(gameUser1);
-        playService.buildCity(gameUser1.getUser(), "1", "3");
+        params = new HashMap<String, String>();
+        params.put("nodeId", "3");
+
+        playService.processAction(GameUserActionCode.BUILD_CITY, gameUser1.getUser(), "1", params);
 
         assertEquals(2, gameUser1.getAchievements().getDisplayVictoryPoints());
     }
 
+    @Test
+    public void shouldPassWhenBuildCityInMainStageOnlyAfterDiceThrown() throws PlayException, GameException{
```

Comment:
the idea was to show in one test case that user can build city ONLY after dice is thrown, but ok, I have splited it to 2 differnt test cases